ui/raidboss/data/06-ew/raid/p2n.ts

```diff
@@ -0,0 +1,178 @@
+        }
+        if (matches.id === '67F7') {
+          if (bodyHeading === 'north')
+            return output.w!();
+          if (bodyHeading === 'west')
+            return output.s!();
+          if (bodyHeading === 'south')
+            return output.e!();
+          if (bodyHeading === 'east')
+            return output.n!();
+        }
+        if (matches.id === '67F9') {
+          if (bodyHeading === 'north')
+            return output.e!();
+          if (bodyHeading === 'west')
+            return output.n!();
+          if (bodyHeading === 'south')
+            return output.w!();
+          if (bodyHeading === 'east')
+            return output.s!();
+        }
+      },
+      outputStrings: {
+        n: Outputs.north,
+        e: Outputs.east,
+        w: Outputs.west,
+        s: Outputs.south,
+        ne: Outputs.northeast,
+        nw: Outputs.northwest,
+        se: Outputs.southeast,
+        sw: Outputs.southwest,
+      },
+    },
+    {
+      id: 'P2N Sewage Deluge',
+      type: 'StartsUsing',
+      netRegex: NetRegexes.startsUsing({ id: '67F6', source: 'Hippokampos', capture: false }),
+      alertText: (_data, _matches, output) => output.text!(),
+      outputStrings: {
+        text: {
+          en: 'Aoe--Get on grid',
+        },
+      },
+    },
+    {
+      // Spread aoe marker on some players, not all
+      id: 'P2N Tainted Flood',
+      type: 'StartsUsing',
+      netRegex: NetRegexes.startsUsing({ id: '6809', source: 'Hippokampos' }),
+      condition: (data, matches) => matches.target === data.me,
+      response: Responses.spread(),
+    },
+    {
+      // Drops aoe zones beneath you -> run to dodge (on everyone)
+      id: 'P2N Sewage Erruption',
+      type: 'StartsUsing',
+      netRegex: NetRegexes.startsUsing({ id: '680D', source: 'Hippokampos', capture: false }),
+      suppressSeconds: 5,
+      response: Responses.spread(),
+    },
+    {
+      id: 'P2N Predatory Sight',
+      type: 'StartsUsing',
+      netRegex: NetRegexes.startsUsing({ id: '680A', source: 'Hippokampos', capture: false }),
+      delaySeconds: 3,
+      response: Responses.doritoStack(),
+    },
+    {
+      // Raidwide knockback -> dont get knocked into slurry
+      id: 'P2N Shockwave',
+      type: 'StartsUsing',
+      netRegex: NetRegexes.startsUsing({ id: '6807', source: 'Hippokampos', capture: false }),
+      alertText: (_data, _matches, output) => output.text!(),
+      outputStrings: {
+        text: {
+          en: 'Knockback--Stay on grid',
+        },
+      },
+    },
+    {
+      // Aoe from head outside the arena
+      id: 'P2N Dissociation',
+      type: 'StartsUsing',
+      netRegex: NetRegexes.startsUsing({ id: '6806', source: 'Hippokampos' }),
+      alertText: (_data, matches, output) => {
+        const xcord = parseFloat(matches.x);
+        if (xcord === 110)
+          return output.w!();
+        if (xcord === 90)
+          return output.e!();
+      },
```

Comment:
Ah, this is nice. Generally it's best to avoid using `getCombatants` where possible, since it's not replayable in the emulator, but I wasn't sure whether there was nonsense with actor movements between the `startsUsing` and `ability` lines that would require it here. Glad this is workable!

Reply:
I started out like that because the log line's positions for Spoken Cataract are completely wrong but turns out in this case the log position is fine.

I guess it would be necessary to find the npc that will later cast the spell to call it out earlier but I've testetd that there is ample time to adjust after the current call.